sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -603,7 +603,13 @@ object DateTimeUtils {
    */
   private[this] def getYearAndDayInYear(daysSince1970: SQLDate): (Int, Int) = {
     // add the difference (in days) between 1.1.1970 and the artificial year 0 (-17999)
-    val daysNormalized = daysSince1970 + toYearZero
+    var  daysSince1970Tmp = daysSince1970
+    // In history,the period(5.10.1582 ~ 14.10.1582) is not exist
```

Comment:
I have tested like this:
`val dav = Date.valueOf("1582-10-14")
  val date = new Date(dav.getTime);
  println(date.toString)`
the output is:`1582-10-24`
`val dav = Date.valueOf("1582-10-05")
  val date = new Date(dav.getTime);
  println(date.toString)`
the output is:`1582-10-15`
so, i think we can not use `1582-10-5` ~ `1582-10-14`